rfc/store-deliver.md

```diff
@@ -0,0 +1,310 @@
+
+```json
+{
+  "ran": "bafy...storeAdd",
+  "out": {
+    "ok": {
+      "status": "upload",
+      "url": "https://...",
+      "link": "bag...",
+      "allocated": 1234
+    }
+  },
+  "fx": {
+    "join": null,
+    "fork": []
+  },
+}
+```
+
+Note that content requested to store, may already be stored. If that is the case, the service may issue receipt with `"status": "done"`.
+
+On their schedule, the agent can post the bytes using the presigned URL. However, both the agent and the service never sign a receipt stating that this content is now stored. This is where `store/deliver` and `store/confirm` can help. In short, we can see `store/deliver` as a way for the agent to sign that they did the work, and `store/confirm` for the service to sign that the bytes were received, stored and can now be retrieved.
+
+While not relevant for verifiability, it is also important pointing out that the service MAY need to run some data computation, issue billing events or write indexes for retrievability asynchronously. Today, web3.storage operated store service relies on "S3 Bucket events" to achieve this, but it should not be assumed that all stores web3.storage will rely on will provide such features.
+
+## New capabilities
+
+### `store/deliver` capability
+
+After the agent handled the asynchronous task of posting the bytes, the agent can invoke `store/deliver` to notify the service that the bytes were delivered. This invocation in the future MAY include a challenge for the service based on the posted data.
+
+```json
+{
+  "can": "store/deliver",
+  "with": "did:key:abc...",
+  "nb": {
+    "link": "bag...",
+    "url": "https://..."
+  }
+}
+```
+
+### `store/confirm` capability
+
+Wwhen the service handles the invocation of `store/deliver`, and potentially, after some asynchronous computations/verifications, the service invokes `store/confirm` in order to issue a signed receipt that the flow is now finished.
+
+```json
+{
+  "can": "store/confirm",
+  "with": "did:key:abc...",
+  "nb": {
+    "link": "bag...",
+    "url": "https://..."
+  }
+}
+```
+
+## Common flows
+
+Aiming to introduce verifiability into content storing, as well as to pave the way for a future proof of delivery, `store/deliver` capability is proposed. It would also make services not tied with store put events to execute side effects.
+
+To offer verifiability, `store/add` invocation should be associated with `store/deliver` and `store/confirm` via [effect]s. `store/deliver` would be invoked by the client to notify the service that content was stored, and the the service should also sign a confirmation that it is true via `store/confirm`.
+
+### `store/deliver` of not previously delivered content
```

Comment:
It may be a good idea to put this into a different namespace so that `with` is our DID not user DID, and `store/add` could delegate capability to invoke it. That would also enable use to perform the task e.g. on some time schedule etc...

General point is space DID here is not a resource, it's our bucket that is a resource.